SQL/optimisations_2017-02-19.sql

```diff
@@ -0,0 +1,206 @@
+, CHANGE COLUMN `adminonly` `adminonly` TINYINT(1) UNSIGNED NOT NULL
+, CHANGE COLUMN `createdby_ckey` `createdby_ckey` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `createdby_ip` `createdby_ip` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `for_trialmin` `for_trialmin` VARCHAR(32) NULL DEFAULT NULL
+, CHANGE COLUMN `dontshow` `dontshow` TINYINT(1) UNSIGNED NOT NULL;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`poll_textreply`
+ CHANGE COLUMN `replytext` `replytext` VARCHAR(2048) NOT NULL
+, ADD COLUMN `ipTEMP` INT UNSIGNED NOT NULL AFTER `ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`poll_textreply`
+ SET `ip` = IF(`ip` IS NULL OR `ip` = '','0',`ip`);
+UPDATE `feedback`.`poll_textreply`
+ SET `ipTEMP` = INET_ATON(`ip`);
+SET SQL_SAFE_UPDATES = 1;
+ALTER TABLE `feedback`.`poll_textreply`
+ DROP COLUMN `ip`
+, CHANGE COLUMN `ipTEMP` `ip` INT(10) UNSIGNED NOT NULL;
+COMMIT;
+
+START TRANSACTION;
+ALTER TABLE `feedback`.`poll_vote`
+ CHANGE COLUMN `ckey` `ckey` VARCHAR(32) NOT NULL
+, ADD COLUMN `ipTEMP` INT UNSIGNED NOT NULL AFTER `ip`;
+SET SQL_SAFE_UPDATES = 0;
+UPDATE `feedback`.`poll_vote`
+ SET `ip` = IF(`ip` IS NULL OR `ip` = '','0',`ip`);
+UPDATE `feedback`.`poll_vote`
+ SET `ipTEMP` = INET_ATON(`ip`);
+SET SQL_SAFE_UPDATES = 1;
+ALTER TABLE `feedback`.`poll_vote`
+ DROP COLUMN `ip`
+, CHANGE COLUMN `ipTEMP` `ip` INT(10) UNSIGNED NOT NULL;
+COMMIT;
+
+/*----------------------------------------------------
+
+These queries are to be run after the above.
+These indexes are designed with only the codebase queries in mind.
+You may find it helpful to modify or create your own indexes if you utilise additional queries for other services.
+
+----------------------------------------------------*/
+
+ALTER TABLE `feedback`.`ban`
+ ADD INDEX `idx_ban_checkban` (`ckey` ASC, `bantype` ASC, `expiration_time` ASC, `unbanned` ASC, `job` ASC)
+, ADD INDEX `idx_ban_isbanned` (`ckey` ASC, `ip` ASC, `computerid` ASC, `bantype` ASC, `expiration_time` ASC, `unbanned` ASC)
+, ADD INDEX `idx_ban_count` (`id` ASC, `a_ckey` ASC, `bantype` ASC, `expiration_time` ASC, `unbanned` ASC);
+
+ALTER TABLE `feedback`.`ipintel`
+ ADD INDEX `idx_ipintel` (`ip` ASC, `intel` ASC, `date` ASC);
+
+ALTER TABLE `feedback`.`library`
+ ADD INDEX `idx_lib_id_del` (`id` ASC, `deleted` ASC)
+, ADD INDEX `idx_lib_del_title` (`deleted` ASC, `title` ASC)
+, ADD INDEX `idx_lib_search` (`deleted` ASC, `author` ASC, `title` ASC, `category` ASC);
+
+ALTER TABLE `feedback`.`messages`
+ ADD INDEX `idx_msg_ckey_time` (`targetckey` ASC, `timestamp` ASC)
+, ADD INDEX `idx_msg_type_ckeys_time` (`type` ASC, `targetckey` ASC, `adminckey` ASC, `timestamp` ASC)
+, ADD INDEX `idx_msg_type_ckey_time_odr` (`type` ASC, `targetckey` ASC, `timestamp` ASC);
+
+ALTER TABLE `feedback`.`player`
+ ADD INDEX `idx_player_cid_ckey` (`computerid` ASC, `ckey` ASC)
+, ADD INDEX `idx_player_ip_ckey` (`ip` ASC, `ckey` ASC);
```

Comment:
I don't I ever had that one (and if I did I nuked my DB when testing so it's gone either way); details? There should be a unique on ckey unless this is different.